packages/anoncreds/src/formats/LegacyIndyCredentialFormatService.ts

```diff
@@ -0,0 +1,603 @@
+    if (!credentialAttributes) {
+      throw new CredentialProblemReportError(
+        `Missing required credential attribute values on credential record with id ${credentialRecord.id}`,
+        { problemCode: CredentialProblemReportReason.IssuanceAbandoned }
+      )
+    }
+
+    const anonCredsIssuerService =
+      agentContext.dependencyManager.resolve<AnonCredsIssuerService>(AnonCredsIssuerServiceSymbol)
+
+    const credentialOffer = offerAttachment?.getDataAsJson<AnonCredsCredentialOffer>()
+    const credentialRequest = requestAttachment.getDataAsJson<AnonCredsCredentialRequest>()
+
+    if (!credentialOffer || !credentialRequest) {
+      throw new AriesFrameworkError('Missing indy credential offer or credential request in createCredential')
+    }
+
+    const { credential, credentialRevocationId } = await anonCredsIssuerService.createCredential(agentContext, {
+      credentialOffer,
+      credentialRequest,
+      credentialValues: convertAttributesToCredentialValues(credentialAttributes),
+    })
+
+    if (credential.rev_reg_id) {
+      credentialRecord.metadata.add<AnonCredsCredentialMetadata>(AnonCredsCredentialMetadataKey, {
+        credentialRevocationId: credentialRevocationId,
+        revocationRegistryId: credential.rev_reg_id,
+      })
+    }
+
+    const format = new CredentialFormatSpec({
+      attachId,
+      format: INDY_CRED,
+    })
+
+    const attachment = this.getFormatData(credential, format.attachId)
+    return { format, attachment }
+  }
+
+  /**
+   * Processes an incoming credential - retrieve metadata, retrieve payload and store it in the Indy wallet
+   * @param options the issue credential message wrapped inside this object
+   * @param credentialRecord the credential exchange record for this credential
+   */
+  public async processCredential(
+    agentContext: AgentContext,
+    { credentialRecord, attachment }: FormatProcessCredentialOptions
+  ): Promise<void> {
+    const credentialRequestMetadata = credentialRecord.metadata.get<AnonCredsCredentialRequestMetadata>(
+      AnonCredsCredentialRequestMetadataKey
+    )
+
+    const registryService = agentContext.dependencyManager.resolve(AnonCredsRegistryService)
+    const anonCredsHolderService =
+      agentContext.dependencyManager.resolve<AnonCredsHolderService>(AnonCredsHolderServiceSymbol)
+
+    if (!credentialRequestMetadata) {
+      throw new CredentialProblemReportError(
+        `Missing required request metadata for credential with id ${credentialRecord.id}`,
+        { problemCode: CredentialProblemReportReason.IssuanceAbandoned }
+      )
+    }
+
+    const anonCredsCredential = attachment.getDataAsJson<AnonCredsCredential>()
+
+    // We can use the same registry for the credential definition and revocation registry as they MUST have the same issuerId
+    const registry = registryService.getRegistryForIdentifier(agentContext, anonCredsCredential.cred_def_id)
```

Comment:
Hmm, I don't think there is anything stopping you from registering an object twice on different ledger, so it might be possible. Although a very very very very insignificant use case I would say. 

I don't think we should have this case.

Reply:
It is something that should be verified by the AnonCreds implementation (now indy-sdk later anoncreds rs). 
I've updated the code here to not make this assumption, and we can handle it in anoncreds and just throw an error there. I think that is cleaner / better

Reply:
Can you make an issue for that check? We can pick it up there.